src/main/java/httpConnectionProcessing/HttpConnectionToProcess.java

```diff
@@ -26,22 +28,28 @@ public void execute() {
     }
   }
 
-  private IHandler getHandler() throws BadConnectionException {
-    HttpRequest request;
-    IHandler handler;
+  private HttpRequest getRequest() throws BadConnectionException {
     try {
-      request = httpRequestBuilder.getNextRequest(client.clientSocketInput());
-      handler = handlerRouter.selectHandler(request);
+      return httpRequestBuilder.getNextRequest(client.clientSocketInput());
     }
     catch (InvalidHttpRequestException e) {
       e.printStackTrace();
+      return null;
+    }
+  }
+
+  private IHandler getHandler(HttpRequest request) throws BadConnectionException {
+    IHandler handler;
+    if (request == null) {
```

Comment:
Yeah - why is `request` ever null? If it's null, some class above this already has this knowledge so maybe do something different there?